rotkehlchen/db/snapshots.py

```diff
@@ -18,41 +21,63 @@ class DBSnapshot:
     def __init__(self, db_handler: DBHandler) -> None:
         self.db = db_handler
 
-    def get_timed_balances(self, timestamp: Timestamp) -> List[DBAssetBalance]:
-        """Retrieves the timed_balances from the db for a given timestamp"""
+    def get_timed_balances(
+        self,
+        timestamp: Timestamp,
+        main_currency: Asset,
+    ) -> List[DBAssetBalance]:
+        """Retrieves the timed_balances from the db for a given timestamp
+           Sets the usd_value to the value in the main currency
+        """
         balances_data = []
         cursor = self.db.conn.cursor()
         timed_balances_result = cursor.execute(
             'SELECT category, time, amount, currency, usd_value FROM timed_balances '
             'WHERE time=?', (timestamp,),
         )
+        main_currency_price = PriceHistorian.query_historical_price(
+            from_asset=A_USD,
+            to_asset=main_currency,
+            timestamp=timestamp,
+        )
         for data in timed_balances_result:
             balances_data.append(
                 DBAssetBalance(
                     category=BalanceType.deserialize_from_db(data[0]),
                     time=data[1],
                     amount=data[2],
                     asset=Asset(data[3]),
-                    usd_value=data[4],
+                    usd_value=str(FVal(data[4]) * main_currency_price),
                 ),
             )
         return balances_data
 
-    def get_timed_location_data(self, timestamp: Timestamp) -> List[LocationData]:
-        """Retrieves the timed_location_data from the db for a given timestamp"""
+    def get_timed_location_data(
+        self,
+        timestamp: Timestamp,
+        main_currency: Asset,
+    ) -> List[LocationData]:
+        """Retrieves the timed_location_data from the db for a given timestamp
+           Sets the usd_value to the value in the main currency
+        """
         location_data = []
         cursor = self.db.conn.cursor()
         timed_location_data = cursor.execute(
             'SELECT time, location, usd_value FROM timed_location_data '
             'WHERE time=?',
             (timestamp,),
         )
+        main_currency_price = PriceHistorian.query_historical_price(
```

Comment:
Yeah, I've added exception catching but how do I notify the user? If I understand correctly, user is only notified when `export` function returns `False` which means that export has totally failed.
I have 2 variants in mind on how to do it:
1) Return False value to notify user and ask them to set another main_currency
2) Return True with a message, but this will require modification of the `rest.py` and perhaps of the fronted
Also if we ask user to add a manual price, how will they do it? As I see, it will require extra field on the frontend where user will input their manual price and thus looks like too much work to handle a case that won't happen often, or am I wrong?

Reply:
You can call the `MessageAggregator` with `msg_aggregator.add_error(` to notify the user with an error. Probably saying that price for timestamp x and asset x is not available and this row was skipped.

For manual prices we already have the option to do this

![image](https://user-images.githubusercontent.com/5068010/164417666-8d4f30a8-a679-4cd9-bcf5-0d48dbee4a77.png)

Reply:
Did it, thanks!